spring-boot-cli/src/it/java/org/springframework/boot/cli/infrastructure/CommandLineInvoker.java

```diff
@@ -187,6 +193,35 @@ public int await() throws InterruptedException {
 			return this.process.waitFor();
 		}
 
+		public void destroy() {
+			if (this.process.isAlive()) {
+				this.process.destroyForcibly();
+			}
+		}
+
+		public boolean isAlive() {
+			return this.process.isAlive();
+		}
+
+		public boolean waitForStarted() {
+			Instant started = Instant.now();
+			for (; ;) {
```

Comment:
`while(ChronoUnit.MILLIS.between(started, Instant.now()) < TIMEOUT)` looks more sensible to me.